tests/test_pythia8.py

```diff
@@ -177,3 +186,32 @@ def test_gp():
 def test_gg():
     evt = run_collision(100 * GeV, "gamma", "gamma")
     assert len(evt) > 2
+
+
+def test_cache_file():
+    evt_kin = CenterOfMass(10 * GeV, "p", "p")
+    m = Pythia8(evt_kin, seed=1, cache=True)
+
+    config_lines = "\n".join(m._config)
+    assert "MultipartonInteractions:reuseInit = 3" in config_lines
+
+    cache_file_path = os.path.join(
+        _cache_base_dir(), "Pythia8", "cache_2212_2212_10000.mpi"
+    )
+    assert os.path.exists(cache_file_path)
```

Comment:
If a set of default cache files are provided in iamdata, this test would just check if these files are there or not. This has to be made sure that this test generates something that's not in the default files.